src/ResourceManager.h

```diff
@@ -22,7 +22,7 @@ class ResourceManager
 	std::vector<std::string> GetAllFilenamesOfType(const std::string& extension, bool accessArchives = true);
 
 	// Returns an empty string if file is not located in an archive file in the ResourceManager's working directory.
-	std::string FindContainingArchiveFile(const std::string& filename);
+	std::string FindContainingArchive(const std::string& filename);
```

Comment:
I prefer short names. Though here we're dealing with objects that represent archives. We need to be clear if we're returning the object that represents the archive and can provide a stream to the contained files, or if we're just returning the path to an archive file on disk that contains the given filename.

To add to the confusion, the base class name for controlling archives is called `ArchiveFile`. What if we renamed this method to `FindContainingArchivePath`.

Reply:
I'd be hesitant to consider a return value as part of function documentation. The return value is not part of the function signature. You can not overload based on the return value. If you ever want to write a method that returns the archive object, that would be very relevant. You'd be forced to choose a different, and likely not ideal name for such a new method.

----

Returning the archive filename with any relative directory portion stripped strikes me as a design flaw. In particular, if you load two archives with the same filename, but placed in different folders, there would be no way to distinguish them. I didn't realize the path was stripped until you pointed it out just now. That's actually contrary to what I would have expected. I would have expected it to return a relative path, with the resource root directory as the base. For archives that appear in the root resource folder, they would then appear as just a filename. For archives in a subfolder, you would see the subfolder as part of the path.

Reply:
If it isn't intuitive, then I'll update as suggested.